app.go

```diff
@@ -0,0 +1,423 @@
+		return
+	}
+
+	if task.ID == "" {
+		notFound(writer, "not found")
+		return
+	}
+
+	log.Info(task)
+
+	jsoned, err := json.Marshal(task)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	writer.Write(jsoned)
+
+	return
+}
+
+func (a *AsyncTasksApp) DeleteByIdRequest(writer http.ResponseWriter, r *http.Request) {
+	var (
+		id string
+		ok bool
+		v  = mux.Vars(r)
+	)
+
+	if id, ok = v["id"]; !ok {
+		badRequest(writer, "No ID in URL")
+		return
+	}
+
+	log.Infof("Fetching async task %s", id)
+
+	tx, err := a.db.BeginTx(context.TODO(), nil)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	defer tx.tx.Rollback()
+
+	task, err := tx.GetTask(id)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	if task.ID == "" {
+		notFound(writer, "not found")
+		return
+	}
+
+	err = tx.DeleteTask(id)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	tx.tx.Commit()
+	return
+}
+
+func (a *AsyncTasksApp) GetByFilterRequest(writer http.ResponseWriter, r *http.Request) {
+	var (
+		v = r.URL.Query()
+
+		filters = TaskFilter{
+			IDs:       v["id"],
+			Types:     v["type"],
+			Statuses:  v["status"],
+			Usernames: v["username"],
+		}
+		start_date_since  = v["start_date_since"]
+		start_date_before = v["start_date_before"]
+		end_date_since    = v["end_date_since"]
+		end_date_before   = v["end_date_before"]
+		null_end          = v["include_null_end"]
+	)
+
+	if len(null_end) > 0 {
+		filters.IncludeNullEnd = true
+	}
+
+	for _, startdate := range start_date_since {
+		parsed, err := time.Parse(time.RFC3339Nano, startdate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.StartDateSince = append(filters.StartDateSince, parsed)
+	}
+
+	for _, startdate := range start_date_before {
+		parsed, err := time.Parse(time.RFC3339Nano, startdate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.StartDateBefore = append(filters.StartDateBefore, parsed)
+	}
+
+	for _, enddate := range end_date_since {
+		parsed, err := time.Parse(time.RFC3339Nano, enddate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.EndDateSince = append(filters.EndDateSince, parsed)
+	}
+
+	for _, enddate := range end_date_before {
+		parsed, err := time.Parse(time.RFC3339Nano, enddate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.EndDateBefore = append(filters.EndDateBefore, parsed)
+	}
+
+	tx, err := a.db.BeginTx(context.TODO(), nil)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	defer tx.tx.Rollback()
+
+	tasks, err := tx.GetTasksByFilter(filters)
+
+	log.Info(tasks)
+
+	jsoned, err := json.Marshal(tasks)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	writer.Write(jsoned)
+
+	return
+}
+
+func (a *AsyncTasksApp) CreateTaskRequest(writer http.ResponseWriter, r *http.Request) {
+	var rawtask AsyncTask
+
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 104857600))
```

Comment:
Magic number?

Reply:
Arbitrary number, basically. It's 100 MiB which I considered a reasonable upper limit on the size of one of these requests in order to not end up running over memory usage. I'll split it out into a constant, maybe.